server/game/GameActions/ReturnCardToDeck.js

```diff
@@ -0,0 +1,26 @@
+const GameAction = require('./GameAction');
+
+class ReturnCardToDeck extends GameAction {
+    constructor() {
+        super('returnCardToDeck');
+    }
+
+    canChangeGameState({ card }) {
+        return card.location !== 'draw deck';
+    }
+
+    createEvent({ card, allowSave = true, bottom = false }) {
+        let params = {
+            card: card,
+            allowSave: allowSave,
+            automaticSaveWithDupe: true,
+            bottom: bottom
+        };
+        return this.event('onCardReturnedToDeck', params, event => {
+            event.cardStateWhenMoved = card.createSnapshot();

```

Comment:
Use the `snapshotName` property you introduced a while back instead. 😄

Reply:
fixed that and also merged in the master so the whole snapshotName stuff is working